game-app/game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/MainPanel.java

```diff
@@ -110,6 +115,22 @@ public class MainPanel extends JPanel {
     add(buttonsPanel, BorderLayout.SOUTH);
     setPreferredSize(initialSize);
     updatePlayButtonState();
+    instance = this;
+  }
+
+  public static void loadSaveFile(final Path file) {
+    // This may be called at any time, including during start up.
+    SwingUtilities.invokeLater(
+        () -> {
+          // If the MainPanel is up already, load the save file.
+          if (instance != null) {
+            instance.gameSelectorPanel.loadSaveFile(file);
+            return;
+          }
+
+          // Otherwise "recurse" to invokeLater() to try again.
+          loadSaveFile(file);
```

Comment:
Thanks for thinking about this more. I had it this way since it's the simplest solution, but agree with your concerns.

It's true that some future change may result in a different start up sequence where `instance` is not set - or not set potentially for a long time. The result of that would be that this code would fail "silently" by repeatedly posting the task and not getting anywhere.

I'm weary of making a maximum iteration count since it's really dependent on how long start up takes and how busy the EDT is (how many times we get to run during start up). A "better" solution is to instead introduce a delay - don't post right away, but run after 100ms. This would avoid having too many such iterations and we could reason about how much time has elapsed.

The main downside of this solution is the code would be more complex. It doesn't look like we have a helper somewhere for `invokeLaterWithDelay()`. It would have to be something like this: https://stackoverflow.com/questions/54459559/run-on-edt-after-delay

WDYT? I can add such a helper and use it here as well as logic to throw an exception if we didn't reach the goal in some amount of time. Let me know if you prefer that - the main downside is the extra code complexity.

Reply:
Ahh, we have `Timers.executeAfterDelay()`. I went ahead and implemented the above and also added a time limit check to prevent infinite recursion.